src/FreqWindow.cpp

```diff
@@ -1034,9 +1062,11 @@ void FrequencyPlotDialog::Recalc()
       if (IsShown())
          blocker.emplace(this);
       wxYieldIfNeeded();
-
-      mAnalyst->Calculate(alg, windowFunc, mWindowSize, mRate,
-         mData.get(), mDataLen,
+      Floats tempFloatsArray;
+      tempFloatsArray.reinit(numSamples);
```

Comment:
Even numbers that do fit in a size_t can be excessive and run out of memory.  There is still a bound of 64M samples and multiply that by four to get bytes.  Will there always be a quarter gig free?

So maybe a try-catch of `bad_alloc` could happen around the allocation, like what I lately did to bullet-proof the Mixer Board.

Reply:
Yep. `Recalc` then also must return a boolean, because the follow-up code that shows the result otherwise crashes. That's not hard, commit upcoming.